bot/cogs/fun.py

```diff
@@ -193,24 +206,87 @@ def __init__(self, bot: Bot):
         Returns a random quotation from the #quotes channel.
         A user can be specified to return a random quotation from that user.
         """
+        conn = await asyncpg.connect()
         quote_channel = self.bot.get_channel(QUOTES_CHANNEL_ID)
-        quotes = self.bot.quotes
 
         if member is None:
-            message_id = choice(quotes[choice(list(quotes.keys()))])
+            message_id = await conn.fetchval(
+                "select quote_id from quotes order by random() limit 1;"
+            )  # fetchval returns first value by default
         else:
-            user_quotes = quotes[f'{member.name}#{member.discriminator}']
-            if not user_quotes:
-                await ctx.send("No quotes from that user.")
+            message_id = await conn.fetchval(
+                "select quote_id from quotes where author_id=$1 order by random() limit 1;",
+                member.id
+            )
+            if message_id is None:
+                await ctx.send("No quotes for that user.")
                 return
-            message_id = choice(user_quotes)
 
+        await conn.close()
         message = await quote_channel.get_message(message_id)
-        await ctx.send(embed=message.embeds[0])
+        if message.author.id == QUOTES_BOT_ID:
+            await ctx.send(embed=message.embeds[0])
+        elif len(message.attachments) > 0:
+            image_url = message.attachments[0].url
+            quote = Embed()
+            quote.set_image(url=image_url)
+            await ctx.send(embed=quote)
+        else:
+            await ctx.send(message.content)
+
+    async def add_quote_to_db(
+        self, conn: asyncpg.connection.Connection, quote: Message
+    ):
+        if quote.author.id == QUOTES_BOT_ID:
```

Comment:
I'd suggest commenting this bit, since it's not entirely clear what's going on

Reply:
done.